packages/fxa-payments-server/src/routes/Product/index.stories.tsx

```diff
@@ -31,22 +37,126 @@ function init() {
         ]
       }} />
     ))
+
+  storiesOf('routes/Product/page load', module)
+    .add('profile loading', () => (
+      <ProductRoute routeProps={{
+        ...MOCK_PROPS,
+        profile: { loading: true, error: null, result: null }
+      }} />
+    ))
+    .add('profile error', () => (
+      <ProductRoute routeProps={{
+        ...MOCK_PROPS,
+        profile: {
+          loading: false,
+          result: null,
+          error: new APIError({ code: 500, message: 'Internal Server Error' }),
+        }
+      }} />
+    ))
+    .add('customer loading', () => (
+      <ProductRoute routeProps={{
+        ...MOCK_PROPS,
+        customer: { loading: true, error: null, result: null }
+      }} />
+    ))
+    .add('customer error', () => (
+      <ProductRoute routeProps={{
+        ...MOCK_PROPS,
+        customer: {
+          loading: false,
+          result: null,
+          error: new APIError({ code: 500, message: 'Internal Server Error' }),
+        }
+      }} />
+    ))
+    .add('plans loading', () => (
+      <ProductRoute routeProps={{
+        ...MOCK_PROPS,
+        plans: { loading: true, error: null, result: null }
+      }} />
+    ))
+    .add('plans error', () => (
+      <ProductRoute routeProps={{
+        ...MOCK_PROPS,
+        plans: {
+          loading: false,
+          result: null,
+          error: new APIError({ code: 500, message: 'Internal Server Error' }),
+        }
+      }} />
+    ))
+  
+  storiesOf('routes/Product/payment failures', module)
+    .add('card declined', () => (
+      <ProductRoute routeProps={{
+        ...FAILURE_PROPS,
+        createSubscriptionStatus: {
+          result: null,
+          loading: false,
+          error: {
+            code: 'card_declined',
+            message: 'Your card has insufficient funds.',
+          }
+        }
+      }} />
+    ))
+    .add('miscellaneous', () => (
+      <ProductRoute routeProps={{
+        ...FAILURE_PROPS,
+        createSubscriptionStatus: {
+          result: null,
+          loading: false,
+          error: {
+            code: '',
+            message: 'Payment server request failed.',
+            params: ''
+          }
+        }
+      }} />
+    ))
+    .add('stripe.createToken() fails on submit', () => {
```

Comment:
I don't see any error in this one?

![image](https://user-images.githubusercontent.com/44390/59469226-d3206700-8df9-11e9-9333-c83bd11f08a8.png)

Reply:
Yeah, this one's hard to cause automatically - you have to actually fill out the form and hit submit. It mocks out the stripe API client and exercises what happens when that fails. Was debating making the sidebar description longer to describe that.